app/src/main/java/io/homeassistant/companion/android/widgets/media_player_controls/MediaPlayerControlsWidget.kt

```diff
@@ -0,0 +1,479 @@
+                )
+                setOnClickPendingIntent(
+                    R.id.widgetMediaPlaceholder,
+                    PendingIntent.getBroadcast(
+                        context,
+                        appWidgetId,
+                        updateMediaIntent,
+                        PendingIntent.FLAG_UPDATE_CURRENT
+                    )
+                )
+                setOnClickPendingIntent(
+                    R.id.widgetPlayPauseButton,
+                    PendingIntent.getBroadcast(
+                        context,
+                        appWidgetId,
+                        playPauseIntent,
+                        PendingIntent.FLAG_UPDATE_CURRENT
+                    )
+                )
+
+                if (showSkip) {
+                    setOnClickPendingIntent(
+                        R.id.widgetPrevTrackButton,
+                        PendingIntent.getBroadcast(
+                            context,
+                            appWidgetId,
+                            prevTrackIntent,
+                            PendingIntent.FLAG_UPDATE_CURRENT
+                        )
+                    )
+                    setOnClickPendingIntent(
+                        R.id.widgetNextTrackButton,
+                        PendingIntent.getBroadcast(
+                            context,
+                            appWidgetId,
+                            nextTrackIntent,
+                            PendingIntent.FLAG_UPDATE_CURRENT
+                        )
+                    )
+                } else {
+                    setViewVisibility(R.id.widgetPrevTrackButton, View.GONE)
+                    setViewVisibility(R.id.widgetNextTrackButton, View.GONE)
+                }
+
+                if (showSeek) {
+                    setOnClickPendingIntent(
+                        R.id.widgetRewindButton,
+                        PendingIntent.getBroadcast(
+                            context,
+                            appWidgetId,
+                            rewindIntent,
+                            PendingIntent.FLAG_UPDATE_CURRENT
+                        )
+                    )
+                    setOnClickPendingIntent(
+                        R.id.widgetFastForwardButton,
+                        PendingIntent.getBroadcast(
+                            context,
+                            appWidgetId,
+                            fastForwardIntent,
+                            PendingIntent.FLAG_UPDATE_CURRENT
+                        )
+                    )
+                } else {
+                    setViewVisibility(R.id.widgetRewindButton, View.GONE)
+                    setViewVisibility(R.id.widgetFastForwardButton, View.GONE)
+                }
+            }
+        }
+    }
+
+    private suspend fun retrieveMediaPlayerImageUrl(entityId: String): String? {
+        val entity = integrationUseCase.getEntities().find { e -> e.entityId.equals(entityId) }
```

Comment:
This can be an expensive call, I know we use it elsewhere but we should consider either in this PR or in the near future add a way to get a single entity rather than all of the entities.

Reply:
I 100% agree. I saw that the Static widget was using it and was incredibly surprised it hadn't been more targeted. I'll see if I can add the code to call just the single entity as part of the next set of changes in this PR.

Reply:
Implemented this change in d923e8c.  Obviously, I had to add an additional GET call in the integration repository.  There are a few things in there that work but I'm not sure if they are the best method possible (specifically ```SingleEntityResponse```) that I would appreciate you looking at extra-closely.